lib/actions/wallet.js

```diff
@@ -0,0 +1,190 @@
+const assert = require('bsert');
+const { getClient } = require('@bpanel/bpanel-utils');
+
+const { actionsLogger: logger } = require('../helpers');
+
+const {
+  SET_WALLETS,
+  USER_SELECT_WALLET,
+  SET_WALLET_INFO,
+  USER_SELECT_MULTISIG_PROPOSAL,
+  SET_MULTISIG_PROPOSAL_MTX,
+  SET_MULTISIG_WALLETS,
+  SET_MULTISIG_WALLET_INFO,
+} = require('../constants');
+
+function setWallets(wallets, walletType) {
+  let type;
+  if (walletType === 'multisig')
+    type = SET_MULTISIG_WALLETS;
+  else
+    type = SET_WALLETS;
+
+  return {
+    type,
+    payload: { wallets },
+  }
+};
+
+function setSelectWallet(walletId, walletType) {
+  let type;
+  if (walletType === 'multisig')
+    type = USER_SELECT_MULTISIG_WALLET;
+  else
+    type = USER_SELECT_WALLET;
+
+  return {
+    type,
+    payload: { walletId },
+  }
+};
+
+function setWalletInfo(walletId, info, walletType) {
+  let type;
+  if (walletType === 'multisig')
+    type = SET_MULTISIG_WALLET_INFO;
+  else
+    type = SET_WALLET_INFO;
+
+  return {
+    type,
+    payload: { info, walletId },
+  }
+};
+
+// app wide...
+function setSelectProposal(proposalId) {
+  return {
+    type: USER_SELECT_MULTISIG_PROPOSAL,
+    payload: { proposalId },
+  };
+};
+
+function setProposalMTX(walletId, proposalId, mtx) {
+  return {
+    type: SET_MULTISIG_PROPOSAL_MTX,
+    payload: { walletId, proposalId, mtx },
+  }
+};
+
+
+/**
+ * Get wallets using wallet client
+ * @param {string} - type
+ * @dispatches
+ * @returns {void}
+ */
+function getWallets(type) {
+  return async dispatch => {
+    const client = getClient();
+
+    logger.debug(`getting ${type ? type : 'standard'} wallets`);
+    let wallets = [];
+    let multisigWallets = [];
+
+    switch (type) {
+      case undefined:
+      case 'standard':
+        wallets.push(...(await client.wallet.getWallets()));
+        break;
+      case 'multisig':
+        multisigWallets.push(...(await client.multisig.getWallets()));
+        break;
+      case 'all':
+        wallets.push(...(await client.wallet.getWallets()));
+        multisigWallets.push(...(await client.multisig.getWallets()));
+        break;
+      default:
+        logger.debug(`invalid argument getWallets(${type})`);
+    }
+
+    if (wallets.length)
+      dispatch(setWallets(wallets))
+
+    if (multisigWallets.length)
+      dispatch(setWallets(multisigWallets, 'multisig'));
+  }
+};
+
+
+/**
+ * User select wallet, fetch wallet info
+ * @param {string} - walletId
+ * @param {string} - type
+ * @dispatches
+ * @returns {void}
+ */
+function selectWallet(walletId, type) {
+  return async dispatch => {
+    const client = getClient();
+    logger.debug(`selecting ${type ? `${type} `: ''} wallet ${walletId}`);
+    dispatch(setSelectWallet(walletId, type));
+    let info;
+    if (type === 'multisig')
```

Comment:
The fact that they are in 2 different databases behind two different http servers makes it difficult. Unless we write a client that abstracts that away

Reply:
Sigh, yeah I know. It was something that we were originally trying to do with the Multisig client actually, kind of acting as a wrapper in front of the wallet client that could then direct requests according to the type.

Reply:
I'm mostly doing it to keep 2 different places in the store, one for normal wallets and one for multisig wallets, so that it is very easy to display only one of each type or all of them. Probably could keep them all in the same place and `filter` each time, but it felt easier to do it this way. Do you have a strong opinion on either way of doing it?

Reply:
My comment was more on having to check the type and use a different client based on that type. It would be nice if you could just `getInfo` and the backend would figure out what type it is (especially since getInfo is basically the same for either type, and the default wallet bclient supports multisig types anyway). I know that's external to this, I just thought that this extra complexity you have to account for highlights this issue. 

As for keeping them separate in the store, I don't really have an opinion on that as I haven't thought about what will be necessary for the UI. In the actual view, when listing wallets for example, the UI won't care about type, but when showing interfaces once a wallet is selected, it will be different. But I haven't thought about the best way to structure the data to accommodate for that is, so whatever you think makes sense is fine by me.